src/main/java/org/zalando/nakadi/service/EventPublisher.java

```diff
@@ -42,28 +47,34 @@
     private final EventTypeCache eventTypeCache;
     private final PartitionResolver partitionResolver;
     private final Enrichment enrichment;
+    private final TimelineSync timelineSync;
 
     @Autowired
     public EventPublisher(final TopicRepository topicRepository,
                           final EventTypeCache eventTypeCache,
                           final PartitionResolver partitionResolver,
                           final Enrichment enrichment,
-                          final NakadiSettings nakadiSettings) {
+                          final NakadiSettings nakadiSettings,
+                          final TimelineSync timelineSync) {
         this.topicRepository = topicRepository;
         this.eventTypeCache = eventTypeCache;
         this.partitionResolver = partitionResolver;
         this.enrichment = enrichment;
         this.nakadiSettings = nakadiSettings;
+        this.timelineSync = timelineSync;
     }
 
     public EventPublishResult publish(final String events, final String eventTypeName, final Client client)
-            throws NoSuchEventTypeException, InternalNakadiException {
-        final EventType eventType = eventTypeCache.getEventType(eventTypeName);
+            throws NoSuchEventTypeException, InternalNakadiException, EventTypeTimeoutException {
+
+        Closeable publishingCloser = null;
         final List<BatchItem> batch = BatchFactory.from(events);
+        try {
+            publishingCloser = timelineSync.workWithEventType(eventTypeName, nakadiSettings.getTimelineWaitTimeoutMs());
```

Comment:
I wanted to do it like that in the beginning but it doesn't work as `workWithEventType` throws exception :(